### PR TITLE
[DataGrid] Fix row click propagation when nesting another grid inside of a grid cell

### PR DESCRIPTION
**Pull request by cherniavskii**
Fixes https://github.com/mui/mui-x/issues/8042

Before: https://codesandbox.io/s/empty-leftpad-7mpf4o?file=/demo.js
After: https://codesandbox.io/s/peaceful-haibt-snddly?file=/demo.js

### CODE REVIEW

File: packages/grid/x-data-grid/src/components/GridRow.tsx [hunk 1]
```diff
@@ -218,6 +218,10 @@ const GridRow = React.forwardRef<
 
   const publishClick = React.useCallback(
     (event: React.MouseEvent<HTMLDivElement>) => {
+      // See https://github.com/mui/mui-x/issues/8042
+      if (!apiRef.current.rootElementRef?.current?.contains(event.target as HTMLElement)) {
```
Comment (cherniavskii) [inline packages/grid/x-data-grid/src/components/GridRow.tsx:222]: `event.stopPropagation()` is an option too, but I'm not sure if it's any better

Comment (MBilalShafi) [reply packages/grid/x-data-grid/src/components/GridRow.tsx:222]: If the goal is also to stop event bubbling, I think `event.stopPropagation` might be a better choice as compared to `return;` which causes event bubbling as depicted in this [codesandbox](https://codesandbox.io/s/bitter-sun-zqh5zg?file=/src/Bar.js). (Comment out `event.stopPropagation()` and uncomment `return` to see log printed in the parent component)

Comment (cherniavskii) [reply packages/grid/x-data-grid/src/components/GridRow.tsx:222]: When the nested grid is in the portal, we cannot rely on the DOM structure to get the parent cell that contains the nested grid.
I've opened https://github.com/mui/mui-x/pull/9741 to solve the original pain reported in the issue, I haven't noticed any issues with propagation with or without Portal

**Comment by cherniavskii** [comment]
Closing it in favor of https://github.com/mui/mui-x/pull/9741

**Pull request closed without merging**